app/models/line_item.rb

```diff
@@ -0,0 +1,24 @@
+class LineItem < ApplicationRecord
+
+  belongs_to :reservation
+  belongs_to :branch_product
+  has_one :product, through: :branch_product
+
+  validates :quantity, :buying_price_in_cents, presence: true
+
+  before_validation :set_price
```

Comment:
you changed before_create to before_validation. this changes the behavior as this will run on update as well which you didn't want.Fix

Reply:
Yes sir fixed.